### PR TITLE
BFD-1696 Fix Jenkins AWS authentication timeouts

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-1696](https://jira.cms.gov/browse/BFD-1696)

**User Story or Bug Summary:**

Allow workflows in Jenkins to authenticate and re-authenticate as necessary.

---

### What Does This PR Do?

In brief, the changes here will allow workflows defined in Jenkins to authenticate **and** re-authenticate as necessary. This **DOES NOT** extend the 60-minute session duration for the `cloudbees-jenkins` role. Instead, we request a new session from a _clean_ environment, assuming the _internal_ `cloudbees-jenkins` role from the _external_ `cbc-bfd` role for every authentication attempt.

First. Because these `withCredentials` blocks update environment variables, we effectively _poison_ the environment for subsequent role-assumption requests _unless_ we are careful to unset those variables the next time we attempt to assume a given role. Here, we unset the appropriate variables each time we make role assumption requests–the default region variable remains untouched, although we **might want to consider storing and restoring the original value before and after these blocks execute.**

Second. While extensively evaluating this solution, I found it helpful to clarify the sessions with meaningful names taken from the Jenkins jobs where the sessions were requested, i.e `bfd-multibranch-and-multistage-pipeline` rather than the overly generic `session`.

Last. I've addressed the following warning:
> Warning: A secret was passed to "sh" using Groovy String interpolation, which is insecure.
>		 Affected argument(s) used the following variable(s): [awsAssumeRole]
>		 See https://jenkins.io/redirect/groovy-string-interpolation for details.

#### Summarized Changes

* encapsulate Jenkins `withCredentials` DSL blocks inside `withEnv` DSL blocks to prevent attempted re-authentication with _potentially_ expired session credentials associated with the _internal_ `cloudbees-jenkins` role; instead, ensure the _external_ `cbc-bfd` role is always used to assume the `cloudbees-jenkins` role by cleaning environment variables in each authentication attempt
* clarify the `--role-session-name` in role assumption to facilitate debugging with AWS Cloudwatch/Cloudtrail
* fix warnings for inappropriate interpolation syntax on shell command for role assumption

#### Future Work
* consider storing and restoring the original value for `AWS_DEFAULT_REGION` before and after `withCredentials` blocks to future issues related to Cloudbees Jenkins homed in us-east-1 attempting to authenticate in unintended regions in _potential_ DR scenarios
* potentially re-factor and re-use authentication logic by defining a method for these purposes
* consider reformatting lengthier lines in the shelled-out `aws` cli commands

### What Should Reviewers Watch For?

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist
I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — DISMISSED** [review]
much better with ENV vars...thx

File: Jenkinsfile [hunk 1]
```diff
@@ -219,14 +222,17 @@ try {
...
-						}
+                        withEnv(['AWS_ACCESS_KEY_ID=','AWS_SECRET_ACCESS_KEY=','AWS_SESSION_TOKEN=']) {
+                            withCredentials([string(credentialsId: 'bfd-aws-assume-role', variable: 'awsAssumeRole')]) {
+								awsCredentials = sh(returnStdout: true, script: 'aws sts assume-role --role-arn "$awsAssumeRole" --role-session-name bfd-multibranch-and-multistage-pipeline --output text --query Credentials').trim().split(/\s+/)
+                                env.AWS_DEFAULT_REGION = 'us-east-1'
+                                env.AWS_ACCESS_KEY_ID = awsCredentials[0]
+                                env.AWS_SECRET_ACCESS_KEY = awsCredentials[2]
+                                env.AWS_SESSION_TOKEN = awsCredentials[3]
+                            }
+                        }
```
Comment (christrotteradhoc) [inline Jenkinsfile:235]: we lost the proper indentation on this block

Comment (mjburling) [reply Jenkinsfile:235]: Yikes. Let me address that. Thanks!

**Review by mackec1 — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]
Great work tracking this down!

**Review by keithdadkins — APPROVED** [review]

**Pull request merged**